### PR TITLE
Exclude openssl related tests from opal suite.

### PR DESCRIPTION
**Pull request by iliabylich**

### CODE REVIEW

File: core/marshal/shared/load.rb [hunk 1]
```diff
@@ -760,27 +760,29 @@
...
+      end
 
-    it "raises ArgumentError when the local class is a regular object" do
-      data = "\004\bd:\020UserDefined\0"
+      it "raises ArgumentError when the local class is a regular object" do
+        data = "\004\bd:\020UserDefined\0"
 
-      lambda { Marshal.send(@method, data) }.should raise_error(ArgumentError)
+        lambda { Marshal.send(@method, data) }.should raise_error(ArgumentError)
+      end
```
Comment (eregon) [inline core/marshal/shared/load.rb]: This is not related to OpenSSL, is it?
Why are tags/filters not good enough?

**Comment by iliabylich** [comment]
@eregon Is opal the only platform supported by mspec that can't work with openssl? If yes - I'm ok with it. I thought that openssl support detection can be organized as a more abstract filter, not just for opal.

**Comment by eregon** [comment]
Most likely not, so these examples should either be changed to not use OpenSSL stuff or create their data locally so they can easily be excluded.

**Comment by eregon** [comment]
Is a literal constant like OpenSSL::X509::Name in non-executed code a problem for Opal?

**Comment by iliabylich** [comment]
@eregon Well, now I understand that it's not related to openssl directly, it's here because openssl has some `T_DATA` type inside (that is used in this test). Using some different class in this test will not give any benefits: all platforms except of opal support it.

**Comment by iliabylich** [comment]
There's no high-level constant OpenSSL in Opal

**Comment by eregon** [comment]
Right, I was asking if Opal can parse (and compile to JS) a file where it meets constants it does not know about.
Seems it can:
```
$ bundle exec bin/opal -e 'if false; p OpenSSL; end; p :ok'
"ok"
```

**Comment by iliabylich** [comment]
@eregon Yes, everything is fine when the code is not invoked. But I invoke it in my patch to Opal and it raises a load error on `require` (and then a name error on resolving a constant name). Simulating a class hierarchy in Opal might fix an error, but that's not a solution.

**Comment by eregon** [comment]
OK, I'll work on moving it or using another class which is less problematic.
Thanks for reporting the issue!

**Comment by iliabylich** [comment]
@eregon Thank you. However I'm not sure whether ruby stdlib has such class. It must be available without explicit `require`-ing and it must have a C pointer inside :smile:

**Comment by eregon** [comment]
Dir has it (a `Dir*` in C). So I changed the spec to use that in 73c4c70b2ddff7db73429819b1d36ad2de51146c.
That should fix the issue :smile:

**Comment by iliabylich** [comment]
@eregon Thanks!

**Comment by elia** [comment]
:+1:
